proxy.go

```diff
@@ -0,0 +1,333 @@
+package veneur
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"os"
+	"strconv"
+	"sync"
+	"syscall"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/Sirupsen/logrus"
+	raven "github.com/getsentry/raven-go"
+	"github.com/hashicorp/consul/api"
+	"github.com/pkg/profile"
+	"github.com/stripe/veneur/discovery"
+	"github.com/stripe/veneur/samplers"
+	"github.com/stripe/veneur/trace"
+	"github.com/zenazn/goji/bind"
+	"github.com/zenazn/goji/graceful"
+	"stathat.com/c/consistent"
+
+	"goji.io"
+	"goji.io/pat"
+)
+
+type Proxy struct {
+	statsd *statsd.Client
+	sentry *raven.Client
+
+	Hostname               string
+	ForwardDestinations    *consistent.Consistent
+	TraceDestinations      *consistent.Consistent
+	Discoverer             discovery.Discoverer
+	ConsulForwardService   string
+	ConsulTraceService     string
+	ConsulInterval         time.Duration
+	ForwardDestinationsMtx sync.Mutex
+	TraceDestinationsMtx   sync.Mutex
+	HTTPAddr               string
+	HTTPClient             *http.Client
+
+	enableProfiling bool
+}
+
+func NewProxyFromConfig(conf ProxyConfig) (ret Proxy, err error) {
```

Comment:
I'm not a huge fan of `ret` as a default return value name - something like `p` would be more idiomatic.